### PR TITLE
prepare monorepo

### LINKED ISSUE #2288: Introduce Monorepo
Before starting Variables M2, we want to introduce a monorepo using `Turborepo`. This will allow us to separate the plugin's core functions into smaller pieces we can then reuse as individual packages.

In this first step, just move everything to a figma-plugin package folder in the monorepo and keep things as they are. We can then incrementally extract pieces into dedicated isolated packages.

### PR DESCRIPTION
**Pull request by six7**
Fixes #2288

Introduces Monorepo (basic) to better allow us split things into smaller packages.

This PR is best reviewed commit by commit.

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "$schema": "https://turbo.build/schema.json",
+  "globalDependencies": ["**/.env.*local"],
+  "pipeline": {
+    "build": {
+      "dependsOn": ["^build"],
+      "outputs": [".next/**", "!.next/cache/**"]
```
Comment (SorsOps) [inline turbo.json:7]: This probably needs dist for this to help it cache

File: turbo.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    },
+    "lint": {},
+    "dev": {
+      "cache": false,
+      "persistent": true
```
Comment (SorsOps) [inline turbo.json:12]: This sometimes causes an issue if it leaves a node process remaining for dependencies. We removed it to promote consistency in dev.

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -154,15 +154,15 @@ jobs:
 
       # 4: Build package
       - name: build:transform
-        run: npm run build-transform
+        run: cd packages/tokens-studio-for-figma && npm run build-transform
```
Comment (SorsOps) [inline .github/workflows/node.js.yml:157]: probably better to use ` working-directory` for this

File: turbo.json (hunk 1 above)
Comment (six7) [reply turbo.json]: Added `dist/**` in latest commit

File: turbo.json (hunk 2 above)
Comment (six7) [reply turbo.json]: Removed that line in latest commit

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (six7) [reply .github/workflows/node.js.yml]: Changed to working-directory for all instances where I cd'd before in latest commit

**Pull request closed without merging**